test/integration/stamp_editor_spec.mjs

```diff
@@ -1097,4 +1098,47 @@ describe("Stamp Editor", () => {
       }
     });
   });
+
+  describe("No auto-resize", () => {
+    let pages;
+
+    beforeAll(async () => {
+      pages = await loadAndWait("empty.pdf", ".annotationEditorLayer", 67);
+    });
+
+    afterAll(async () => {
+      await closePages(pages);
+    });
+
+    it("must check that a stamp editor isn't resizing itself", async () => {
+      // Run sequentially to avoid clipboard issues.
+      const editorSelector = getEditorSelector(0);
+
+      for (const [, page] of pages) {
+        await switchToStamp(page);
+
+        await copyImage(page, "../images/firefox_logo.png", 0);
+        await page.waitForSelector(editorSelector);
+        await waitForSerialized(page, 1);
+      }
+
+      await Promise.all(
+        pages.map(async ([browserName, page]) => {
+          const getDims = () =>
+            page.evaluate(sel => {
+              const bbox = document.querySelector(sel).getBoundingClientRect();
+              return `${bbox.width}::${bbox.height}`;
+            }, editorSelector);
+          const initialDims = await getDims();
+          for (let i = 0; i < 50; i++) {
+            // eslint-disable-next-line no-restricted-syntax
+            await waitForTimeout(10);
```

Comment:
Is there no better way to handle this, since we've seen repeatedly that using "arbitrary" timeouts often leads to intermittent test failures?
(Ideally we should get rid of that helper function completely.)

Reply:
In the issue it's stated that this happens when the zoom level is changed, so perhaps we can use that property to immediately trigger the bug? I agree that waiting for a more-or-less arbitrary time should be avoided unless there is really no other way (since especially on the slower Windows bot that's bound to fail intermittently).

Reply:
The zoom level doesn't have to be changed: in patch #18698 we changed the way to compute the size of the page and so we can have some differences (depending on the zoom level) when computing:
https://github.com/mozilla/pdf.js/blob/8e5f06d77d5ebc9ff66d5d57f99109b553ff4e8a/src/display/editor/editor.js#L665
This difference combined with https://github.com/mozilla/pdf.js/blob/8e5f06d77d5ebc9ff66d5d57f99109b553ff4e8a/src/display/editor/stamp.js#L556 (which is removed in this patch) just changes constantly the size (which is updated thanks to a ResizeObserver).

That said, here the waitTimeout isn't a problem: we aren't waiting for something but we're just waiting for nothing to happen !
So of course if the resize is triggered after ~500ms we'll be stuck, but I think it's very unlikely.

This is the second time we've this bug (see #16797) and I don't want to have it a third time, so a test is absolutely required.
That said, if you can imagine a better way to test that stuff, please tell me.

Reply:
> [...] we aren't waiting for something but we're just waiting for nothing to happen !

Can you please add a (short) comment about this, to aid future readers of this code?